libraries/system.rb

```diff
@@ -3,18 +3,18 @@ module System
     class FormFactor
       attr_reader :machine_model
 
-      def initialize(machine_model)
-        @machine_model = machine_model
+      def initialize(hardware)
+        @machine_model = hardware.nil? ? nil : hardware['machine_model']
```

Comment:
Nitpick: If you use empty string instead of `nil`, you won't need the `return false if @machine_model.nil?` duplication below.

Reply:
Actually if it's possible for `node['hardware']` to be `nil` then this would fail. If that's possible, we can still remove the duplication here with something like this:

```ruby
def initialize(hardware)
  @machine_model = (hardware.nil? || !hardware['machine_model']) ? '' : hardware['machine_model']
```

Reply:
I'll wait to refactor this until we put a PR into fauxhai.